File: storydoc-frontend/src/app/api/models/ui-scenario-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ScreenShotTimeLineItemDto } from './screen-shot-time-line-item-dto';
import { ScreenshotCollectionSummaryDto } from './screenshot-collection-summary-dto';
import { StoryDocSummaryDto } from './story-doc-summary-dto';
import { TimeLineModelCoordinate } from './time-line-model-coordinate';
import { UiScenarioId } from './ui-scenario-id';
export interface UiScenarioDto {
  associatedCollections?: Array<ScreenshotCollectionSummaryDto>;
  id?: UiScenarioId;
  name?: string;
  screenshots?: Array<ScreenShotTimeLineItemDto>;
  state?: 'CREATED' | 'READY';
  storyDocSummary?: StoryDocSummaryDto;
  timeLineModelCoordinate?: TimeLineModelCoordinate;
}

File: storydoc-backend/src/main/java/io/storydoc/server/ui/app/UIScenarioDTO.java
package io.storydoc.server.ui.app;

import io.storydoc.server.storydoc.app.dto.StoryDocSummaryDTO;
import io.storydoc.server.storydoc.domain.ArtifactState;
import io.storydoc.server.timeline.domain.TimeLineModelCoordinate;
import io.storydoc.server.ui.domain.UIScenarioId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UIScenarioDTO {

    private StoryDocSummaryDTO storyDocSummary;

    private UIScenarioId id;

    private String name;

    private ArtifactState state;

    private TimeLineModelCoordinate timeLineModelCoordinate;

    private List<ScreenShotTimeLineItemDTO> screenshots;

    private List<ScreenshotCollectionSummaryDTO> associatedCollections;

}
